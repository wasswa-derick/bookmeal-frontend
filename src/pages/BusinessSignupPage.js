import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import BusinessSignupForm from "../components/forms/BusinessSignupForm";
import { registerBusiness } from "../actions/auth";
import AppNavBar from "../components/NavBar";
/**
 * @class Signup
 * @extends {React.Component}
 */
export class BusinessSignupPage extends React.Component {
  /**
   * @param {object} data to submit
   * @returns {Promise} promise to the caller
   */
  handleSubmit = data =>
    this.props
      .registerBusiness(data)
      .then(() => this.props.history.push("/login"));

  /**
   * @returns {any} renders a page
   */
  render() {
    return (
      <div>
        <AppNavBar />
        <div className="container login-panel">
          <div className="card auth-card">
            <div className="card-body">
              <h3 className="card-title title">Create Your Business Account</h3>
              <BusinessSignupForm handleSubmit={this.handleSubmit} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

BusinessSignupPage.propTypes = {
  registerBusiness: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};
export default connect(null, { registerBusiness })(BusinessSignupPage);
