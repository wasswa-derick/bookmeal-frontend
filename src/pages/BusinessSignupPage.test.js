import React from "react";
import { shallow } from "enzyme";
import { BusinessSignupPage } from "./BusinessSignupPage";

describe("BusinessSignup", () => {
  let wrapper;
  beforeEach(() => {
    const history = {
      push: jest.fn
    };
    wrapper = shallow(
      <BusinessSignupPage registerBusiness={jest.fn} history={history} />
    );
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
