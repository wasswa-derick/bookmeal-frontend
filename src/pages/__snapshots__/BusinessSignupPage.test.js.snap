// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessSignup should render correctly 1`] = `
<div>
  <NavBar />
  <div
    className="container login-panel"
  >
    <div
      className="card auth-card"
    >
      <div
        className="card-body"
      >
        <h3
          className="card-title title"
        >
          Create Your Business Account
        </h3>
        <BusinessSignupForm
          handleSubmit={[Function]}
        />
      </div>
    </div>
  </div>
</div>
`;
